### DISCUSSION TITLE
How to get x,y,z and quaternion by using LPS?

### DISCUSSION THREAD

**Discussion by Wallat1** [bitcraze/discussions#311] [Q&A] [answered]
Hi.

I have started to use the LPS, sice I would like to fly the CF outdoor, for this motive could you explain me how to get the information of x, y, z and of the quaternion  please?, I am using Python.

Regards.

**Comment by knmcguire** [comment, marked as answer]
Hi!

You can use the [Logging variable of StateEstimate ](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/#stateestimate) for that, which is the LPS information fused with the IMU of the crazyflie.

You can read it out with the [CFlib](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/), via these instructions:
- https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/sbs_connect_log_param/
- https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/python_api/#logging
- Example scripts: https://github.com/bitcraze/crazyflie-lib-python/tree/master/examples/logging

Depends on what you want with the information, but you can use the [cfclient logger tab](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/plotter_tab/) to display those as well.

**Reply by Wallat1** [reply]
Hi.

Thank you for your help.

Regards.
